src/transport/transport.h

```diff
@@ -119,6 +119,10 @@ class ConvectionTransport : public TransportBase {
 	 * Initialize solution at zero time.
 	 */
     void zero_time_step() override;
+    /**
+     * 
+     */
+    bool assess_time_constraint(double &time_constraint);
```

Comment:
Dosumentation. 'Assess' have misleading meaning in this context I propose to use 'evaluate' instead.